### ISSUE TITLE
Expected input batch_size to match target batch_size error occurs while bidir=True

### ISSUE THREAD

**Issue by alexwwang** [n-waves/ulmfit-multilingual#23]
I am trying to use cpu to train an `lstm` ulmfit lm with `bidir=true` and the error occurs while the first round in  `fit_one_cycle`.

Concretely, the error happened at `torch.nn.functional.py#Ln1788` in `nll_loss.format(input.size(0), target.size(0)))`. 

The above level shows the program is trying to compute `cross_entropy` in `nn.modules.loss.forward`.

I'm using pytorch-nightly-cpu with fastai installed by conda.

Anyone could provide any hint on how to dig in? Thanks.

**Comment by PiotrCzapla** [comment]
Yeah, there were some changes in pytorch and fastai that caused this issue. I've fixed it in the refactoring branch. (it should work with the latest fastai, I've tested it with n-waves/fastai:ulmfit_multilingual_refactor).

**Comment by PiotrCzapla** [comment]
reopen if you see the error again.

**Comment by alexwwang** [comment]
Great to hear that. Thank you for replying.
Happy new year!
